files/faban/src/cloud/benchflow/collectors/faban.go

```diff
@@ -0,0 +1,98 @@
+package main
+ 
+import (
+    "fmt"
+    "io/ioutil"
+    "net/http"
+    "os"
+    "os/exec"
+    "encoding/json"
+    "log"
+    "bytes"
+    "github.com/benchflow/commons/minio"
+    "github.com/Shopify/sarama"
+    "strconv"
+)
+
+type KafkaMessage struct {
+	SUT_name string `json:"SUT_name"`
+	SUT_version string `json:"SUT_version"`
+	Minio_key string `json:"minio_key"`
+	Trial_id string `json:"trial_id"`
+	Experiment_id string `json:"experiment_id"`
+	Container_id string `json:"container_id"`
+	Total_trials_num int `json:"total_trials_num"`
+	Collector_name string `json:"collector_name"`
+	}
+
+func signalOnKafka(minioKey string, containerID string) {
+	totalTrials, _ := strconv.Atoi(os.Getenv("BENCHFLOW_TRIAL_TOTAL_NUM"))
+	kafkaMsg := KafkaMessage{SUT_name: os.Getenv("SUT_NAME"), SUT_version: os.Getenv("SUT_VERSION"), Minio_key: minioKey, Trial_id: os.Getenv("BENCHFLOW_TRIAL_ID"), Experiment_id: os.Getenv("BENCHFLOW_EXPERIMENT_ID"), Container_id: containerID, Total_trials_num: totalTrials, Collector_name: os.Getenv("BENCHFLOW_COLLECTOR_NAME")}
+	jsMessage, err := json.Marshal(kafkaMsg)
+	if err != nil {
+		log.Printf("Failed to marshall json message")
+		}
+	producer, err := sarama.NewSyncProducer([]string{os.Getenv("KAFKA_HOST")+":9092"}, nil)
+	if err != nil {
+	    log.Fatalln(err)
+	}
+	defer func() {
+	    if err := producer.Close(); err != nil {
+	        log.Fatalln(err)
+	    }
+	}()
+	msg := &sarama.ProducerMessage{Topic: os.Getenv("KAFKA_TOPIC"), Value: sarama.StringEncoder(jsMessage)}
+	partition, offset, err := producer.SendMessage(msg)
+	if err != nil {
+	    log.Printf("FAILED to send message: %s\n", err)
+	    } else {
+	    log.Printf("> message sent to partition %d at offset %d\n", partition, offset)
+	    }
+	}
+ 
+func backupHandler(w http.ResponseWriter, r *http.Request) {
+    query := r.FormValue("trial_id")
+    path := "/app/faban/"+query
```

Comment:
Define a variable where you specify the path were `Faban` stores the data we want to collect, as documented in https://github.com/benchflow/collectors/issues/36.

We just pass the actual `trial_id` that is the name of the folder inside the `output` folder.